dev/dns.go

```diff
@@ -58,33 +59,28 @@ func (d *DNSResponder) handleDNS(w dns.ResponseWriter, r *dns.Msg) {
 	w.WriteMsg(m)
 }
 
+// Serve binds to
 func (d *DNSResponder) Serve(domains []string) error {
 	for _, domain := range domains {
 		dns.HandleFunc(domain+".", d.handleDNS)
 	}
 
 	addr := d.Address
 	if addr == "" {
-		addr = DefaultAddress
+		return fmt.Errorf("invalid port specification in %v", d)
```

Comment:
just bail if the address is invalid. don't fall back on a default that would break `-dns-port` flag usage.